core-processor/src/ext.rs

```diff
@@ -611,13 +612,23 @@ impl EnvExt for Ext {
     }
 
     fn read(&mut self) -> Result<&[u8], Self::Error> {
-        let size = self
+        let max_payload_size: u32 = MAX_PAYLOAD_SIZE
+            .try_into()
+            .expect("statically asserted to be in u32 bounds");
+        let (size, with_error) = self
             .size()?
             .try_into()
-            .map_err(|_| MessageError::IncomingPayloadTooBig)?;
-
+            .map(|v| (v, false))
+            // This must not happen. because max size of the payload
```

Comment:
should we assert it so? :/

Reply:
I'd live without an assertion in the back-end until proper types and refactoring is introduced (discussed with you several weeks ago).

Reply:
done